signac_dashboard/views.py

```diff
@@ -62,41 +74,51 @@ def show_job(dashboard, jobid):
         g.jobs = dashboard._get_job_details([job])
         g.title = g.jobs[0]["title"]
         g.subtitle = g.jobs[0]["subtitle"]
+        session["context"] = "JobContext"
         return dashboard._render_job_view(default_view="grid")
 
 
-def get_file(dashboard, jobid, filename):
-    try:
-        job = dashboard.project.open_job(id=jobid)
-    except KeyError:
-        abort(404, "The job id requested could not be found.")
+def get_file(dashboard, filename, jobid=None):
+    if jobid is not None:
+        try:
+            job_or_project = dashboard.project.open_job(id=jobid)
+        except KeyError:
+            abort(404, "The job id requested could not be found.")
+    else:
+        job_or_project = dashboard.project
+    if job_or_project.isfile(filename):
+        directory = (
+            job_or_project.root_directory()
+            if jobid is None
+            else job_or_project.workspace()
+        )
```

Comment:
The `.fn("")` pattern you used earlier might work here (at the risk of being hard to understand). I have complex feelings on this and maybe avoiding `.fn("")` is the better approach, even though it increases the amount of code to write in several places. Maybe we need a helper function... I don't know. It just feels like a hack. The current implementation is fine.